smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/middlewares/providers/HttpHeaderProvider.kt

```diff
@@ -114,6 +115,8 @@ class HttpHeaderProvider(
                 writer.openBlock("if $memberName != ${member.defaultValue(ctx.symbolProvider)} {", "}") {
                     writer.write("items.add(Header(name: \"$paramName\", value: \$N($memberNameWithExtension)))", SwiftTypes.String)
                 }
+            } else if (inCollection && ctx.model.expectShape(member.target) !is TimestampShape) {
+                writer.write("items.add(Header(name: \"$paramName\", value: quoteHeaderValue(\$N($memberNameWithExtension))))", SwiftTypes.String)
             } else {
                 writer.write("items.add(Header(name: \"$paramName\", value: \$N($memberNameWithExtension)))", SwiftTypes.String)
             }
```

Comment:
**Question**: Why shouldn't timestamps get their value quoted like everything else?

Reply:
Initially I was actually doing the quoting for all values (timestamp and string) and protocol tests were failing.

Then I looked into [kotlin implementation  ](https://github.com/awslabs/smithy-kotlin/blob/f6eeecb2cda7c345bef6f510dc30eac2c0697725/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/protocol/HttpStringValuesMapSerializer.kt#L93-L120) which also quote only for plain string.

Reply:
It seems like this code will quote everything that isn't a timestamp (i.e., `!is TimestampShape`). Have you verified this works with lists of booleans, numbers, etc.?

Reply:
Verified from protocol tests defined at https://github.com/awslabs/smithy[/blob/9af3fab79eeee645e28ca92e0b711cfa17472a5d/smithy-aws-protocol-tests/model/restJson1/http-headers.smithy](https://github.com/awslabs/smithy/blob/9af3fab79eeee645e28ca92e0b711cfa17472a5d/smithy-aws-protocol-tests/model/restJson1/http-headers.smithy

- "X-StringSet": "a, b, c"
- "X-IntegerList": "1, 2, 3",
- "X-BooleanList": "true, false, true"
- "X-TimestampList": "Mon, 16 Dec 2019 23:48:18 GMT, Mon, 16 Dec 2019 23:48:18 GMT"
- "X-EnumList": "Foo, Bar, Baz"

Also, [quoteHeaderValue](https://github.com/awslabs/smithy-swift/blob/jangirg/chore/bump-smithy-version/Packages/ClientRuntime/Sources/Serialization/SerializationUtils/HeaderUtils.swift#L77-L91) doesn't always quote. The string must contains quotable chars which is also translation of [kotlin](https://github.com/awslabs/smithy-kotlin/blob/9e5c5acc1922821b15ceff93385ea6230dce57ff/runtime/protocol/http/common/src/aws/smithy/kotlin/runtime/http/util/HeaderLists.kt#L122-L134) version.

Reply:
Makes sense, thanks!